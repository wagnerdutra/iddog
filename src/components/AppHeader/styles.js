import styled from 'styled-components';

const Header = styled.header`
  background: var(--main-bg-color);
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
  margin: 100px;

  div {
    display: flex;
    p {
      white-space: nowrap;

      span {
        font-weight: bold;
      }
    }

    @media only screen and (max-width: 350px) {
      flex-direction: column;
      align-items: center;
    }
  }

  @media only screen and (max-width: 1050px) {
    flex-direction: column;
    align-items: center;

    div {
      @media only screen and (max-width: 1050px) {
        margin-top: 20px;
      }
    }
  }
`;

const Nav = styled.ul`
  list-style: none;
  display: flex;
  margin-left: 20px;

  li {
    button {
      background: transparent;
      border: none;
      border-bottom: 1px solid;
      font-size: 18px;
      cursor: pointer;
    }
  }
`;

export { Header, Nav };
