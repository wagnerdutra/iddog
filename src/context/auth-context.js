import React from 'react';
import { useAsync } from 'react-async';

import axios from '../services/api';

import { bootstrapApp } from '../utils/bootstrapApp';

import { FullScreenLoading } from '../components/Loading';

const AuthContext = React.createContext();

function AuthProvider(props) {
  const { data = { email: null, token: null }, isPending, reload } = useAsync({
    promiseFn: bootstrapApp
  });

  if (isPending) {
    return <FullScreenLoading />;
  }

  const register = form =>
    axios
      .post(
        '/signup',
        { ...form },
        {
          headers: {
            'Content-Type': 'application/json'
          }
        }
      )
      .then(({ data: { user } }) => {
        window.localStorage.setItem('userData', JSON.stringify(user));
        reload();
        return user;
      });

  const logout = () => {
    window.localStorage.removeItem('userData');
    reload();
  };

  return <AuthContext.Provider value={{ data, register, logout }} {...props} />;
}

function useAuth() {
  const context = React.useContext(AuthContext);
  if (context === undefined) {
    throw new Error(`useAuth must be used within a AuthProvider`);
  }
  return context;
}

export { AuthProvider, useAuth };
