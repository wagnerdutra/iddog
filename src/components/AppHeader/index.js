import React from 'react';

import { useAuth } from 'Root/context/auth-context';
import { useUser } from 'Root/context/user-context';

import Title from 'Components/AppTitle';

import { Header, Nav } from './styles';

const AppHeader = () => {
  const { logout } = useAuth();
  const user = useUser();

  return (
    <Header>
      <Title />
      <div>
        <p>
          Welcome:
          <span>{` ${user}`}</span>
        </p>
        <Nav>
          <li>
            <button onClick={logout} type="button">
              Logout
            </button>
          </li>
        </Nav>
      </div>
    </Header>
  );
};

export default AppHeader;
