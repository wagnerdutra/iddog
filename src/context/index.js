import React from 'react';
import propTypes from 'prop-types';

import { AuthProvider } from './auth-context';
import { UserProvider } from './user-context';

const AppProviders = ({ children }) => (
  <AuthProvider>
    <UserProvider>{children}</UserProvider>
  </AuthProvider>
);

AppProviders.propTypes = {
  children: propTypes.node.isRequired
};

export default AppProviders;
