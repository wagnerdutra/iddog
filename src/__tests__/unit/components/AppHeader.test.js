import React from 'react';

import { render, fireEvent } from 'react-testing-library';

import AppHeader from 'Components/AppHeader';

import { useAuth } from 'Root/context/auth-context';
import { useUser } from 'Root/context/user-context';

const logout = jest.fn();

jest.mock('Root/context/auth-context');
jest.mock('Root/context/user-context');

useUser.mockReturnValue('wagnerdutra1010@gmail.com');
useAuth.mockReturnValue({ logout });

describe('<AppHeader />', () => {
  it('should render correctly', () => {
    const { container } = render(<AppHeader />);
    expect(container).toMatchSnapshot();
  });

  it('should has user email', () => {
    const { getByText } = render(<AppHeader />);
    const email = getByText('wagnerdutra1010@gmail.com');
    expect(email).toBeInTheDocument();
  });

  it('should handle logout click', () => {
    const { getByText } = render(<AppHeader />);
    const button = getByText('Logout');
    fireEvent.click(button);
    expect(logout).toHaveBeenCalledTimes(1);
  });
});
