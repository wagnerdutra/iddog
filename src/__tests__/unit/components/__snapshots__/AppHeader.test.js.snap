// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppHeader /> should render correctly 1`] = `
.c1 {
  -webkit-letter-spacing: 10px;
  -moz-letter-spacing: 10px;
  -ms-letter-spacing: 10px;
  letter-spacing: 10px;
  font-weight: normal;
  white-space: nowrap;
}

.c1 span {
  font-weight: bold;
}

.c0 {
  background: var(--main-bg-color);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  margin: 100px;
}

.c0 div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 div p {
  white-space: nowrap;
}

.c0 div p span {
  font-weight: bold;
}

.c2 {
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: 20px;
}

.c2 li button {
  background: transparent;
  border: none;
  border-bottom: 1px solid;
  font-size: 18px;
  cursor: pointer;
}

@media only screen and (max-width:350px) {
  .c0 div {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media only screen and (max-width:1050px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<div>
  <header
    class="c0"
  >
    <h1
      class="c1"
    >
      THE
      <span>
         IDDOG
      </span>
    </h1>
    <div>
      <p>
        Welcome:
        <span>
           wagnerdutra1010@gmail.com
        </span>
      </p>
      <ul
        class="c2"
      >
        <li>
          <button
            type="button"
          >
            Logout
          </button>
        </li>
      </ul>
    </div>
  </header>
</div>
`;
